File: BOJ/10000 ~ 11000/P10809.java
// https://www.acmicpc.net/problem/10809

import java.util.Scanner;

public class P10809 {
    public static void main(String[] args) {
        char[] alphabet = "abcdefghijklmnopqrstuvwxyz".toCharArray();
        Scanner sc = new Scanner(System.in);
        String word = sc.next();

        for (int i=0; i<alphabet.length; i++) {
            char c = alphabet[i];
            int idx = word.indexOf(c);
            if (i == alphabet.length - 1) {
                System.out.print(idx);
            } else{
                System.out.print(idx + " ");
            }
        }
    }
}

File: BOJ/10000 ~ 11000/P10809.kt
// https://www.acmicpc.net/problem/10809

fun main(args: Array<String>) {
    val word = readln()
    val position = Array(size = 26) { -1 }

    for((idx, c) in word.withIndex()) {
        val offset = c - 'a'
        if (position[offset] == -1) position[offset] = idx
    }

    for ((idx, p) in position.withIndex()) {
        if (idx == position.size - 1) print("$p")
        else print("$p ")
    }
}
